cursive-core/src/views/select_view.rs

```diff
@@ -957,21 +985,22 @@ impl<T: 'static + Send + Sync> View for SelectView<T> {
                 PaletteStyle::Primary
             };
 
-            let x = match printer.size.x.checked_sub(1) {
-                Some(x) => x,
+            let lw = match printer.size.x.checked_sub(self.decorators_width()) {
+                Some(lw) => lw,
                 None => return,
             };
 
             printer.with_style(style, |printer| {
-                // Prepare the entire background
-                printer.print_hline((1, 0), x, " ");
-                // Draw the borders
-                printer.print((0, 0), "<");
-                printer.print((x, 0), ">");
+                // Prepare the label background
+                printer.print_hline((self.decorators[0].width(), 0), lw, " ");
+                // Draw the decorators
+                printer.print((0, 0), &self.decorators[0]);
+                printer.print((self.decorators[0].width() + lw, 0), &self.decorators[1]);
```

Comment:
Can we store `self.decorators[0].width()` in a variable, so we only need to compute it once in this function call?
We'll still compute it an extra time as part of `decorators_width()`, but it might sacrifice some readability to avoid that.